src/AssistantThreadContextStore.ts

```diff
@@ -0,0 +1,88 @@
+import { MessageMetadataEventPayloadObject } from '@slack/types';
+import { AllAssistantMiddlewareArgs, extractThreadInfo } from './Assistant';
+
+export interface AssistantThreadContextStore {
+  get: GetThreadContextFn;
+  save: SaveThreadContextFn;
+}
+
+export interface GetThreadContextFn {
+  (args: AllAssistantMiddlewareArgs): Promise<AssistantThreadContext>;
+}
+
+export interface SaveThreadContextFn {
+  (args: AllAssistantMiddlewareArgs): Promise<void>;
+}
```

Comment:
@seratch I went back and forth on this: namely, the typing of `get` and `save`. I had to be somewhat prescriptive here (instead of just `any` everywhere), but am not sure the current typing selections reflect what a developer might actually do with these functions. Would like to hear your thoughts.

Reply:
I thought that we can narrow the possible values down than the AllAssistantMiddlewareArgs list. This means accepting only SlackEventMiddlewareArgs<'assistant_thread_started'> etc. would sound more natural to me

Reply:
It sounds good in theory, but I'm not sure that's feasible given the current limitations of the type system.

Related, both methods are available for use in all listeners, and the event args are just passed along (as they are to all utils). I'm not following why we would narrow it down further if the event could be any of the three handlers. 

Happy to adjust this, though, if you can help clear that up for me!

Reply:
This function is not a listener/middleware but a utility function (meaning same with say etc.), therefore I don't see the necessity to add extra utility methods like listners/middlware have. That said, if there is a technical challenge with types, it's okay to go with the current code this time.